torchaudio/prototype/ctc_decoder/ctc_decoder.py

```diff
@@ -129,20 +129,24 @@ def __call__(self, emissions: torch.FloatTensor, lengths: Optional[torch.Tensor]
 
         Args:
             emissions (torch.FloatTensor): tensor of shape `(batch, frame, num_tokens)` storing sequences of
-                probability distribution over labels; output of acoustic model
+                probability distribution over labels; output of acoustic model. It must lives on CPU.
             lengths (Tensor or None, optional): tensor of shape `(batch, )` storing the valid length of
-                in time axis of the output Tensor in each batch
+                in time axis of the output Tensor in each batch. It must lives on CPU.
 
         Returns:
             List[List[Hypothesis]]:
                 List of sorted best hypotheses for each audio sequence in the batch.
         """
         assert emissions.dtype == torch.float32
 
+
         B, T, N = emissions.size()
         if lengths is None:
             lengths = torch.full((B,), T)
 
+        assert not emissions.is_cuda
+        assert not lengths.is_cuda
```

Comment:
The logic it self looks good. Couple of suggestions.

1. For a better UX, could you provide an error message that tell users what to do (as opposed to what was wrong)?
2. Please perform the input validation as soon as possible, before any operation is performed. Otherwise the operation before validation could be wasteful
3. This is not a written rule, but `assert` is more fore internal assertion. (although it is used in L140, which I guess was missed at a review time) So can replace it with `if <condition>: raise <something>`?